doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,345 @@
+expertise. It should provide timers which can competently handle the timescales
+of any application.
+
+## Interoperability
+
+RIOT nodes need to communicate reliably with non-RIOT nodes. The protocols need
+to be familiar to them and behave in the way they expect.
+
+RIOT should support open network protocol standards once they have reached a
+certain level of validity and popularity. It should be configurable so that
+users can choose which (optional) areas of specifications they want to include.
+Whatever the configuration, nodes should handle all possible traffic in a
+compliant manner.
+
+## Stability
+
+Nodes shouldn't fail because of RIOT, whatever the platform. Assuming perfect
+hardware and a moderately well-written application, they should be able to run
+indefinitely.
+
+The testing and peer review processes that ensure this should be under
+continuous review and refinement, including automating them where it's
+possible. Error handling should guarantee stability with minimal memory usage.
+
+## Unified APIs
+
+Users don't want to be confused by semantic differences between APIs. RIOT
+provides a similar "look and feel" across all our APIs.
+
+RIOT's interfaces should only differ in the ways that their modules differ.
+Where possible, layers should exist that sit on top of a certain class of
+modules to give an identical interface to the user. Semantic and naming
+conventions should be consistent throughout the system.
+
+## Internet security
+
+RIOT nodes need to be resilient against cyber attacks, for the sake of its
+reputation and the reputation of the IoT.
+
+It should be hard against all the threats a node is likely to experience,
+depending on the situation, including those from computers with many times
+their processing power. Security should be as easy to use as possible while
+still strong. Security flaws should be patched as quickly as possible.
+Convenient updating mechanisms should allow users to apply patches to their
+nodes, wherever they are.
+
+# Acknowledgements
+
+This document follows previous work on documenting RIOT's design priorities [3]
+[4].
+
+# References
+
+[1] [RFC7228 Terminology for Constrained-Node Networks. C. Bormann, M. Ersue,
+A. Keranen. May 2014.](https://tools.ietf.org/html/rfc7228) \
+[2] [“What Is Free Software?” _GNU Operating System_, 15 Dec. 2018,
+www.gnu.org/philosophy/free-sw.en.html.](https://www.gnu.org/philosophy/free-sw.en.html) \
+[3] [Emmanuel Baccelli, Oliver Hahm, Mesut Günes, Matthias Wählisch, Thomas
+C.  Schmidt, "RIOT OS: Towards an OS for the Internet of Things," in
+Proceedings of the 32nd IEEE International Conference on Computer
+Communications (INFOCOM), Poster Session, April
+2013.](https://www.riot-os.org/docs/riot-infocom2013-abstract.pdf) \
+[4] [E. Baccelli, et al. 2018. RIOT: An Open Source Operating System for Low-end
+Embedded Devices in the IoT. The IEEE Internet of Things Journal
+(2018).](http://ilab-pub.imp.fu-berlin.de/papers/bghkl-rosos-18-prepub.pdf)
```

Comment:
This reference is incomplete - please also name all authors, since we are not space-constrained:
E. Baccelli, C. Gündogan, O. Hahm, P. Kietzmann, M. Lenders, H. Petersen, K. Schleiser, T.C. Schmidt, M.
Wählisch: "RIOT: an Open Source Operating System for Low-end Embedded Devices in the IoT",
IEEE Internet of Things Journal, Vol. 5, No. 6, p. 4428- 4440, IEEE, December 2018.